doc/developer/design/20230814_unbilled_replicas.md

```diff
@@ -0,0 +1,203 @@
+# Unbilled replicas
+
+- [#20317](https://github.com/MaterializeInc/materialize/issues/20317)
+
+## Context
+
+Support requires the ability to dynamically add resources to a user's environment
+without incurring substantial costs for the user. For example, when support
+notices that after a version upgrade, Materialize needs additional compute
+resources, support should be able to add such resources for a limited amount of
+time without requiring explicit approval and complicated reimbursement for the
+customer.
+
+We extend Materialize with a concept of unbilled replicas, which server compute
+traffic but are not charged to a customer's account.
+
+To achieve this, we extend clusters and their replicas with concept of replica sets.
+A replica set defines properties shared by a partition of replicas within a cluster.
+We allow internal users to create replica sets with specific billing information to add free or cost-reduced resouces to user environemnts, without changing the structure of a user's deplyoment.
+
+## Goals
+
+We introduce unbilled clusters by the following means:
+* Introduce replica sets.
+* Add a billing property to replica sets.
+* Record replica set properties in audit events.
+
+## Non-Goals
+
+We do not plan to do the following:
+* Allow customers general access to replica sets.
+
+## Overview
+
+We introduce the concept of replica sets, which capture partitions
+of replicas sharing configuration aspects. Replica sets can co-exist, and a default
```

Comment:
"default" is also named `NULL`, "implicit", and "implicit default" in other places here. I don't much care what the name is, but this document should use a single term.